x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/BlobCacheMetrics.java

```diff
@@ -113,22 +114,28 @@ public LongHistogram getCacheMissLoadTimes() {
     /**
      * Record the various cache population metrics after a chunk is copied to the cache
      *
+     * @param blobName The file that was requested and triggered the cache population.
      * @param bytesCopied The number of bytes copied
      * @param copyTimeNanos The time taken to copy the bytes in nanoseconds
      * @param cachePopulationReason The reason for the cache being populated
      * @param cachePopulationSource The source from which the data is being loaded
      */
     public void recordCachePopulationMetrics(
+        String blobName,
```

Comment:
Should this be `@Nullable` or is this information interesting for the indexing tier as well? 
cc @henningandersen

Reply:
Should there be a serverless PR linked to this one? I think this method is only called in serverless currently.

(I think it'd help to have some context to make this call)

I'd err on the side of not making it nullable unless we needed to for some reason?

Reply:
I am not sure I understand the question. It seems the `blobName` must be filled in or we get NPE? And how is it relevant whether it is indexing or search?

I'd probably want any metrics at the blob cache level on both tiers.

Reply:
@nicktindall the serverless PR is linked to this one https://github.com/elastic/elasticsearch-serverless/pull/3534

Reply:
@henningandersen sure, the current implementation would NPE but the parameter is not currently marked as `@Nullable`. 
I would've made the needed changes had we said we only want the metrics at the search tier level. 

Thanks for your insight here, sounds like we're good to go as is.

Reply:
Oh I see it was linked, it's not as apparent on the ES PR as it is on the serverless one.

Yeah this LGTM, consistency between tiers is good. I do wonder if 50 would be considered "high" cardinality though. It would be interesting to see in practice how much it affects the overall number of time series. It worries me that we still don't monitor `agent.events.dropped` (see [here](https://elasticco.atlassian.net/browse/ES-10023))